codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -230,15 +230,14 @@ AsciiString end() throws Http2Exception {
         }
 
         private void append(int i) {
-            try {
-                bytes[index] = (byte) i;
-            } catch (IndexOutOfBoundsException ignore) {
+            if (bytes.length >= index) {
                 // Always just expand by INITIAL_SIZE
                 byte[] newBytes = new byte[bytes.length + initialCapacity];
```

Comment:
I wonder if this is worth re-visiting (followup PR). Perhaps we should be more aggressive at first (double it until a certain threshold is crossed)?